### DISCUSSION TITLE
Is it possible to build dotnet with Clang/LLVM on Windows 10 (hopefully with -O3 and -march:native)?

### DISCUSSION THREAD

**Discussion by MineCake147E** [dotnet/runtime#47450] [General] [answered]
I'm wondering what the difference would be if I replace MSVC with Clang.
I don't have any idea to do so, so I need to know how building dotnet works, and how to change the C compiler for dotnet.

**Comment by huoyaoyuan** [comment, marked as answer]
CMake is used to compile native components.

Windows and non-Windows platforms uses different bootstrap scripts. The Unix script has a switch to pass GCC to CMake. Windows script seems hard-codes MSVC.

> I'm wondering what the difference would be if I replace MSVC with Clang.

As a hosting runtime, ABI and calling convention are extremely important for CLR.  Clang [supports MSVC ABI](https://clang.llvm.org/docs/MSVCCompatibility.html), so it remains possible.

CMake is invoked at src/coreclr/build-runtime.cmd
